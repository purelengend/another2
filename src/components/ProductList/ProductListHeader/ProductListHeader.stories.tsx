import type { Meta, StoryObj } from '@storybook/react'
import ProductListHeader from './index'

const meta = {
  title: 'Practice Two/ProductListHeader Component',
  component: ProductListHeader,
  decorators: [
    Story => (
      <div
        className="storybook-test"
        style={{
          backgroundColor: '#532EE7',
          color: '#FFF',
        }}
      >
        <Story />
      </div>
    ),
  ],

  tags: ['autodocs'],
} satisfies Meta<typeof ProductListHeader>

export default meta
type Story = StoryObj<typeof meta>

export const Primary: Story = {}
