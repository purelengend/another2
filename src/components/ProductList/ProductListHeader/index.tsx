import { memo } from 'react'
import { Box, Heading, Grid, Text } from '@chakra-ui/react'

const ProductListHeader = () => (
  <Box
    textAlign="center"
    fontFamily="Oswald-Regular"
    color="textSecondary"
    margin={{ base: '0 -50px 16px' }}
  >
    <Heading
      as="h2"
      className="product-list-title"
      fontSize={{ base: '30px', md: '48px', lg: '60px' }}
    >
      All Products
    </Heading>
    <Grid gridTemplateColumns={'1fr auto 1fr'} alignItems="center">
      <Text as="span" height={1} backgroundColor="secondary" />
      <Text
        as="span"
        className="maxim"
        width="block"
        px={{ base: 1, md: 6 }}
        my={{ base: 4, lg: 6 }}
        fontSize={{ base: '16px', lg: '24px' }}
      >
        Complete range of products for all your needs
      </Text>
      <Text as="span" height={1} backgroundColor="secondary" />
    </Grid>
  </Box>
)

export default memo(ProductListHeader)
