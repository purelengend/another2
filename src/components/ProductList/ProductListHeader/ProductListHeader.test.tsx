import { render } from '@testing-library/react'
import ProductListHeader from '.'

describe('ProductListHeader Component', () => {
  it('renders the title "All Products"', () => {
    const { getByText } = render(<ProductListHeader />)
    const titleElement = getByText('All Products')
    expect(titleElement).toBeInTheDocument()
  })

  it('renders the maxim text', () => {
    const { getByText } = render(<ProductListHeader />)
    const maximText = getByText('Complete range of products for all your needs')
    expect(maximText).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<ProductListHeader />)
    expect(asFragment()).toMatchSnapshot()
  })
})
