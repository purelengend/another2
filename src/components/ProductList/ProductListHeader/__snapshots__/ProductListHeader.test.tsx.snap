// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListHeader Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-cxazlp"
  >
    <h2
      class="chakra-heading product-list-title css-1rcmjjf"
    >
      All Products
    </h2>
    <div
      class="css-lxhvv8"
    >
      <span
        class="chakra-text css-fimz8h"
      />
      <span
        class="chakra-text maxim css-268hzw"
      >
        Complete range of products for all your needs
      </span>
      <span
        class="chakra-text css-fimz8h"
      />
    </div>
  </div>
</DocumentFragment>
`;
